### PR TITLE
Improve --help ux for Canton sandbox

### LINKED ISSUE #12985: Sandbox help UX is very confusing
`daml sandbox --help` shows the help of the sandbox wrapper
`daml sandbox -- -- --help` shows the help of the underlying Canton runner.

That is super confusing and unintuitive. We should do better.

### PR DESCRIPTION
**Pull request by cocreature**
fixes #12985

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/tools/sandbox.rst [hunk 1]
```diff
@@ -188,8 +188,8 @@ To start Sandbox, run: ``daml sandbox [options] [-c canton.config]``.
 
 To see all the available options, run ``daml sandbox --help``. Note
 that this will show you the options of the Sandbox wrapper around
-Canton. To see options of the underlying Canton runner, use ``daml
-sandbox -- -- --help``.
+Canton. To see options of the underlying Canton runner, use
+``daml sandbox --canton-help``.
```
Comment (stefanobaghino-da) [inline docs/source/tools/sandbox.rst:192]: Should we rather consider going the `git` way and doing an `help` subcommand? I.e. `daml help sandbox`

Comment (cocreature) [reply docs/source/tools/sandbox.rst:192]: I don’t see how that helps. You still have help for the sandbox wrapper and for the underlying canton JAR. Git doesn’t have that issue `git status --help` and `git help status` are the same help text.

Comment (stefanobaghino-da) [reply docs/source/tools/sandbox.rst:192]: > I don’t see how that helps. You still have help for the sandbox wrapper and for the underlying canton JAR. Git doesn’t have that issue `git status --help` and `git help status` are the same help text.

Yes, but it doesn't have `git status --status-help`. I would agree with you if you could have `daml sandbox --help`. Right now as a user I would prefer to see the help text for Canton by typing that. I expect some confusion from users over using `daml sandbox --help` vs. `daml sandbox --canton-help`.

**Review by sofiafaro-da — APPROVED** [review]
LGTM, thank you!

File: docs/source/tools/sandbox.rst (hunk 1 above)
Comment (cocreature) [reply docs/source/tools/sandbox.rst:192]: A few comments to explain why we’re in this situation: Canton itself does not have any special sandbox mode or even a separate JAR. So we need some wrapper that turns Canton into a somewhat usable sandbox by specifying certain options and config files. This wrapper has to make some choices like ports. It also provides some UX niceties that people expect in sandbox and make it easier to migrate like options to turn on static time. The idea here is that in 95% of cases, you only ever need to interact with the wrapper directly and you don’t have to care that there is an underlying Canton JAR.

However, Canton has thousands of options and we clearly don’t want to replicate all of them in the wrapper. So for advanced usages you might need to look at the Canton help.

I fully agree that the UX isn’t great and somewhat leaks how things are implemented internally. I don’t really have a better suggestion though. We can try to parse Canton’s --help output and integrate it but that seems super hacky and is bound to fail in the future. We could make `daml sandbox -- --help` work but I’m not sure that’s really better (it still leaks implementation details of there being an underlying wrapper) and it’s not discoverable. --canton-help has the advantage that it shows up if you just type --help.

**Pull request merged**